Python/codecs.c

```diff
@@ -50,6 +50,26 @@ int PyCodec_Register(PyObject *search_function)
     return -1;
 }
 
+PyObject *
+_PyCodec_Unregister(PyObject *search_function)
+{
+    PyInterpreterState *interp = PyInterpreterState_Get();
+    if (interp->codec_search_path == NULL) {
+        Py_RETURN_NONE;
+    }
+
+    Py_ssize_t n = PyList_Size(interp->codec_search_path);
```

Comment:
Instead of creating a new loop, you can use your loop but call PyList_SetSlice(list, i, i+1, NULL) to remove the item in-place. See list.remove() implementation.

Reply:
make sense, have been updated.